### PR TITLE
[WIP] Faster BigInt Implement via Malachite

### PR DESCRIPTION
**Pull request by qingshi163**
#4797

### CODE REVIEW

File: common/src/int.rs [hunk 1]
```diff
@@ -1,120 +1,631 @@
...
+        // log2(x) = log2(2^n * 2^-n * x) = n + log2(x/2^n)
+        // If we set 2^n to be the greatest power of 2 below x, then x/2^n is in [1, 2), and can
+        // thus be converted into a float.
+        let n = self.bits() as u32 - 1;
+        let frac = Rational::from_integers_ref(self.inner(), Self::from(2).pow(n.into()).inner());
+        let float: f64 = frac.rounding_into(RoundingMode::Down);
+        f64::from(n) + float.log2()
+    }
+
+    pub fn getrandbits<F>(mut k: usize, mut f: F) -> Self
```
Comment (youknowone) [inline common/src/int.rs:388]: is this part of BigInt?

File: common/src/int.rs [hunk 2]
```diff
@@ -1,120 +1,631 @@
...
+    }
+
+    pub fn to_bytes_le(&self) -> (Ordering, Vec<u8>) {
+        (
+            self.sign(),
+            self.inner().unsigned_abs_ref().to_power_of_2_digits_asc(8),
+        )
+    }
+
+    pub fn from_bytes_be(sign: bool, digits: &[u8]) -> Self {
```
Comment (youknowone) [inline common/src/int.rs:501]:
```suggestion
    pub fn from_bytes_be(sign: Sign, digits: &[u8]) -> Self {
```
BigInt will be a part of public API of RustPython.
I wish we avoid boolean for non-boolean-meaning values for public interface as documenting purpose.

File: common/src/int.rs [hunk 3]
```diff
@@ -1,120 +1,631 @@
...
+    From,
+    Shr,
+    Shl,
+    Sum,
+    Product,
+    ShlAssign,
+    ShrAssign,
+)]
+#[from(forward)]
+pub struct BigInt(Integer);
```
Comment (youknowone) [inline common/src/int.rs:53]: It seems you are almost making alternative num-bigint implementation. 
I think making a crate like `malachite-bigint` as a drop-in replacement of `num-bigint` might be worth if you are interested in.

Comment (qingshi163) [reply common/src/int.rs:53]: malachite-bigint is a good idea

**Comment by qingshi163** [comment]
> Can we close in preference to #4952?

sure

**Pull request closed without merging**